android/src/main/java/com/segment/analytics/kotlin/android/plugins/AndroidContextPlugin.kt

```diff
@@ -123,16 +125,43 @@ class AndroidContextPlugin : Plugin {
             emptyJsonObject
         }
 
+        // generate random identifier that does not persist across installations
+        // use it as the default device id in case DRM API failed to generate one.
+        val fallbackDeviceId = UUID.randomUUID().toString()
```

Comment:
is it better to have no device id or like a string of 0s to indicate not collected. and then use the fallback one when we fail to fetch via DRM?

Reply:
yeah. that's a good call out. my consideration is that in the case when DRM does fail, we have a consistent device id before and after (it takes 2s to know it's failed, could have some cached events). but using all 0s to indicate not ready to collect is also a good idea.